test/utils/shallowEquality.spec.js

```diff
@@ -0,0 +1,71 @@
+import expect from 'expect';
+import shallowEqualScalar from '../../src/utils/shallowEqualScalar';
+import shallowEqual from '../../src/utils/shallowEqual';
+
+describe('Utils', () => {
+  // More info: https://github.com/gaearon/redux/pull/75#issuecomment-111635748
+  describe('shallowEqualScalar', () => {
+    it('should compare the given objects shallowly', () => {
+
+      expect(shallowEqualScalar(null, {})).toBe(false);
+      expect(shallowEqualScalar({}, null)).toBe(false);
+      const m = {one: 1};
+
+      expect(shallowEqualScalar([], [])).toBe(true);
+      expect(shallowEqualScalar([], [1, 2])).toBe(false);
+      expect(shallowEqualScalar([3, 4], [3, 4])).toBe(true);
+      expect(shallowEqualScalar([3, 4], [3, 5])).toBe(false);
+      expect(shallowEqualScalar([[]], [[]])).toBe(false);
+      expect(shallowEqualScalar([m], [m])).toBe(false);
+
+      const n = {two: 2};
+
+      expect(shallowEqualScalar(m, n)).toBe(false);
```

Comment:
I'd prefer to put these objects inline into method calls so it's easier to see what's being compared.